transformer_lens/HookedTransformer.py

```diff
@@ -1470,20 +1471,39 @@ def generate(
 
         assert isinstance(tokens, torch.Tensor)
         batch_size, ctx_length = tokens.shape
-        tokens = tokens.to(devices.get_device_for_block_index(0, self.cfg))
+        device = devices.get_device_for_block_index(0, self.cfg)
+        tokens = tokens.to(device)
         if use_past_kv_cache:
             past_kv_cache = HookedTransformerKeyValueCache.init_cache(
                 self.cfg, self.cfg.device, batch_size
             )
         else:
             past_kv_cache = None
 
-        if stop_at_eos and eos_token_id is None:
-            assert (
+        stop_tokens = []
+        pad_token_id = 0
+        if stop_at_eos:
```

Comment:
```suggestion
        if stop_at_eos:
            pad_token_id = 0
```

Reply:
I don't think we want to leave a variable in scope if it's not used at all. Let me know if the type checker isn't happy and we can find a workaround

Reply:
What you propose makes sense, but that leads to the warning `Local variable 'pad_token_id' might be referenced before assignment ` in PyCharm.

Reply:
I took into account all your other comments.